Foundation/Unit.swift

```diff
@@ -248,7 +248,7 @@ open class Dimension : Unit {
     }
 }
 
-open class UnitAcceleration : Dimension {
+final class UnitAcceleration : Dimension {
```

Comment:
I don't think we can take this patch? These subclasses are open in ObjC, and we want as-close-as-possible source parity.

Reply:
(I assume we're trying to work around the fact that Swift 5 prevents us from having a pure Swift class that has the equivalent of `+ (instancetype)baseUnit`?)

Reply:
At this point I'm testing the option for consideration.  If we took it there are other things to consider, such as parity with Objective-C and expectations for users of the APIs.

Reply:
IIRC, subclassing is part of the contract of this API. (I will check tomorrow.)

I have a patch I can PR that moves .baseUnit() to be of Dimension type everywhere, erasing the type but allowing the build to work. I will discuss it with @parkera tomorrow morning.

Reply:
Yeah, and Swift is just highlighting it. :(

I'll circle back before EOD.

Reply:
(It feels like Swift could allow `Self` class methods if they were similar to `required` initializers, wherein it is not valid to express a class that doesn't override the method.)

Reply:
Yes, that sounds like potentially a reasonable language extension that does not exist today.

Reply:
For now, to unblock Swift 5 adoption, we will take this patch.